### PR TITLE
Control constants for STP

### LINKED ISSUE #964: Split constants.h into smaller constants files
If you change something now, you have to literally recompile the entire AI

### PR DESCRIPTION
**Pull request by RobotJente**
Made a new constants file for control constants for STP
I accidentally put the play naming thing in this PR as well, was not intentional. :disappointed:

### CODE REVIEW

File: src/stp/new_skills/Chip.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@ void Chip::onInitialize() noexcept {}
 
 Status Chip::onUpdate(const StpInfo &info) noexcept {
     // Clamp and set chip velocity
-    double chipVelocity = std::clamp(info.getKickChipVelocity(), 0.0, Constants::MAX_KICK_POWER());
+    double chipVelocity = std::clamp(info.getKickChipVelocity(), 0.0, stp::control_constants::MAX_KICK_POWER());
```
Comment (timovdk) [inline src/stp/new_skills/Chip.cpp]: So close hahaha
```suggestion
    double chipVelocity = std::clamp(info.getKickChipVelocity(), 0.0, stp::control_constants::MAX_KICK_POWER);
```

File: test/StpTests/PlayCheckerTests.cpp [hunk 2]
```diff
@@ -3,31 +3,32 @@
...
     uint8_t score(rtt::world_new::World *world) noexcept override { return 0; }
 
     void assignRoles() noexcept override {}
 
     void calculateInfoForRoles() noexcept override {}
 
-    bool isValidPlayToStart(rtt::world_new::World *world) noexcept override { return false; }
-    bool isValidPlayToKeep(rtt::world_new::World *world) noexcept override { return false; }
     bool shouldRoleSkipEndTactic() override { return false; }
+
```
Comment (timovdk) [inline test/StpTests/PlayCheckerTests.cpp]:
```suggestion
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -182,6 +182,8 @@ set(NEW_TACTICS_SOURCES
         ${PROJECT_SOURCE_DIR}/src/stp/new_tactics/Receive.cpp
         ${PROJECT_SOURCE_DIR}/src/stp/new_tactics/ChipAtPos.cpp
         ${PROJECT_SOURCE_DIR}/src/stp/new_tactics/Intercept.cpp
+        ${PROJECT_SOURCE_DIR}/src/stp/new_constants/ControlConstants.cpp
+
```
Comment (timovdk) [inline CMakeLists.txt]:
```suggestion
```

File: include/roboteam_ai/stp/Play.hpp [hunk 4]
```diff
@@ -59,9 +60,9 @@ class Play {
     virtual ~Play() = default;
 
     /**
-     * Default ctor, ensures proper construction of Play
+     * Ctor that constructs a play an assigns its name
```
Comment (timovdk) [inline include/roboteam_ai/stp/Play.hpp]: I think I made a typo, oops
```suggestion
     * Ctor that constructs a play and assigns its name
```

File: include/roboteam_ai/stp/new_constants/ControlConstants.h [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    extern const double DRIBBLER_ANGLE_OFFSET;
+    extern const double FRONT_LENGTH;
+
+}
+
+
+
+
+
+#endif //RTT_CONTROLCONSTANTS_H
```
Comment (timovdk) [inline include/roboteam_ai/stp/new_constants/ControlConstants.h]:
```suggestion
}

#endif //RTT_CONTROLCONSTANTS_H
```

File: src/stp/Skill.cpp [hunk 6]
```diff
@@ -12,6 +12,7 @@
 #include "utilities/Settings.h"
 #include "world_new/World.hpp"
 
+
```
Comment (timovdk) [inline src/stp/Skill.cpp]:
```suggestion
```

File: src/stp/new_constants/ControlConstants.cpp [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    // Robot physical constants
+    constexpr double BALL_RADIUS = 0.0215;
+    constexpr double ROBOT_RADIUS = 0.089;
+    constexpr double ROBOT_RADIUS_MAX = 0.091;
+    const double CENTER_TO_FRONT = sin(DRIBBLER_ANGLE_OFFSET) * ROBOT_RADIUS;
+    const double DRIBBLER_ANGLE_OFFSET = asin(FRONT_LENGTH / 2 / ROBOT_RADIUS);
+    constexpr double FRONT_LENGTH = 0.118;
+}
+
+
```
Comment (timovdk) [inline src/stp/new_constants/ControlConstants.cpp]:
```suggestion
```

File: src/stp/new_tactics/KickAtPos.cpp [hunk 8]
```diff
@@ -52,12 +52,12 @@ StpInfo KickAtPos::calculateInfoForSkill(StpInfo const &info) noexcept {
 /// Determine how fast we should kick for a pass at a given distance
 // TODO: This is bad code full of magic numbers so please refactor at a later stage :)
 double KickAtPos::determineKickForce(double distance, KickChipType desiredBallSpeedType) noexcept {
-    const double maxPowerDist = rtt::ai::Constants::MAX_POWER_KICK_DISTANCE();
+    const double maxPowerDist = stp::control_constants:::MAX_POWER_KICK_DISTANCE;
```
Comment (timovdk) [inline src/stp/new_tactics/KickAtPos.cpp]:
```suggestion
    const double maxPowerDist = stp::control_constants::MAX_POWER_KICK_DISTANCE;
```

File: test/StpTests/PlayCheckerTests.cpp (hunk 2 above)
Comment (timovdk) [inline test/StpTests/PlayCheckerTests.cpp]:
```suggestion
```

File: src/stp/new_skills/Rotate.cpp [hunk 9]
```diff
@@ -16,7 +16,7 @@ Status Rotate::onUpdate(const StpInfo &info) noexcept {
 
     // Clamp and set dribbler speed
     int targetDribblerPercentage = std::clamp(info.getDribblerSpeed(), 0, 100);
-    int targetDribblerSpeed = targetDribblerPercentage / 100.0 * Constants::MAX_DRIBBLER_CMD();
+    int targetDribblerSpeed = targetDribblerPercentage / 100.0 * control_constants::MAX_DRIBBLER_CMD;
```
Comment (jordiluong) [inline src/stp/new_skills/Rotate.cpp]:
```suggestion
    int targetDribblerSpeed = targetDribblerPercentage / 100.0 * stp::control_constants::MAX_DRIBBLER_CMD;
```
consistency

File: src/stp/new_skills/Rotate.cpp [hunk 10]
```diff
@@ -27,7 +27,7 @@ Status Rotate::onUpdate(const StpInfo &info) noexcept {
     publishRobotCommand();
 
     // Check if successful
-    double errorMargin = Constants::GOTOPOS_ANGLE_ERROR_MARGIN() * M_PI;
+    double errorMargin = control_constants::GO_TO_POS_ANGLE_ERROR_MARGIN * M_PI;
```
Comment (jordiluong) [inline src/stp/new_skills/Rotate.cpp]:
```suggestion
    double errorMargin = stp::control_constants::GO_TO_POS_ANGLE_ERROR_MARGIN * M_PI;
```

File: src/stp/new_tactics/DriveWithBall.cpp [hunk 11]
```diff
@@ -55,7 +55,7 @@ bool DriveWithBall::shouldTacticReset(const StpInfo& info) noexcept {
     auto robotAngle = info.getRobot()->get()->getAngle();
     auto ballToRobotAngle = (info.getBall()->get()->getPos() - info.getRobot()->get()->getPos()).angle();
 
-    return fabs(robotAngle + ballToRobotAngle) <= Constants::GOTOPOS_ANGLE_ERROR_MARGIN();
+    return fabs(robotAngle + ballToRobotAngle) <= control_constants::GO_TO_POS_ANGLE_ERROR_MARGIN;
```
Comment (jordiluong) [inline src/stp/new_tactics/DriveWithBall.cpp]:
```suggestion
    return fabs(robotAngle + ballToRobotAngle) <= stp::control_constants::GO_TO_POS_ANGLE_ERROR_MARGIN;
```

**Review by timovdk — APPROVED** [review]

**Review by jordiluong — APPROVED** [review]

File: src/stp/new_tactics/ChipAtPos.cpp [hunk 12]
```diff
@@ -47,6 +47,7 @@ StpInfo ChipAtPos::calculateInfoForSkill(StpInfo const &info) noexcept {
 }
 
 double ChipAtPos::determineChipForce(double distance, KickChipType desiredBallSpeedType) noexcept {
+
```
Comment (RobotJente) [inline src/stp/new_tactics/ChipAtPos.cpp]:
```suggestion
```

**Pull request merged**